### PR TITLE
[flutter_tools] Remove sound null safety flag

### LINKED ISSUE #118810: Remove option to pass unsound null safety flag in flutter build/run commands
Starting in Dart3.beta, the Dart SDK will not support the option to pass in `--no-sound-null-safety flag`, Flutter would also have to change to not support this option when building/running Flutter apps

Please see https://github.com/dart-lang/sdk/issues/50348 for more details regarding the Dart SDK change.

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/118810

Remove both the `--sound-null-safety` and `--null-assertions` flags from the Flutter CLI tool. These will be removed from the Dart SDK in the 3.0 release. Before this change, we were defaulting to `--sound-null-safety` and `--no-null-assertions`, and providing a warning if the user explicitly overrode this that they were going to be deprecated soon.

Note there is no change in this PR related to the `--native-null-assertions` flag; it defaults to on but the user can explicitly disable it, which is the desired behavior per https://github.com/flutter/flutter/issues/118810#issuecomment-1430567784.

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/bootstrap.dart [hunk 1]
```diff
@@ -168,7 +167,6 @@ require.config({
 define("$bootstrapModule", ["$entrypoint", "dart_sdk"], function(app, dart_sdk) {
   dart_sdk.dart.setStartAsyncSynchronously(true);
   dart_sdk._debugger.registerDevtoolsFormatter();
-  dart_sdk.dart.nonNullAsserts($nullAssertions);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/bootstrap.dart:171]: @nshahan is it ok that I deleted this line? Or should I leave this line and hard-code the value (since I deleted the CLI arg in this PR)?

Comment (nshahan) [reply packages/flutter_tools/lib/src/web/bootstrap.dart:171]: Yes, looks like good cleanup. This is totally unneeded as long as you are prevented from passing 
`--no-sound-null-safety` when compiling. It will be defaulted to `false` in the compiled SDK until it is fully cleaned up.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/web/bootstrap.dart:171]: Thanks!

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -118,21 +118,8 @@ class FlutterDevice {
     // used to file a bug, but the compiler will still start up correctly.
     if (targetPlatform == TargetPlatform.web_javascript) {
       // TODO(zanderso): consistently provide these flags across platforms.
-      final String platformDillName;
+      const String platformDillName = 'ddc_outline_sound.dill';
```
Comment (nshahan) [inline packages/flutter_tools/lib/src/resident_runner.dart:121]: Is there cleanup somewhere to remove the unsound .dill file from the packaged flutter release? Not essential but will cut down the overall download size.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/resident_runner.dart:121]: That's a good point, I'll file a tracking issue for this.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/resident_runner.dart:121]: https://github.com/flutter/flutter/issues/122215

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 3]
```diff
@@ -91,12 +91,7 @@ enum HostArtifact {
   webPlatformDart2JSSoundKernelDill,
 
   /// The precompiled SDKs and sourcemaps for web debug builds.
-  webPrecompiledSdk,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/artifacts.dart:94]: Is this cleanup related to the flag removal?

File: packages/flutter_tools/lib/src/reporting/events.dart [hunk 4]
```diff
@@ -248,67 +248,3 @@ class CodeSizeEvent extends UsageEvent {
...
-      }
-      if (packageLanguageVersion != null &&
-          packageLanguageVersion.major >= nullSafeVersion.major &&
-          packageLanguageVersion.minor >= nullSafeVersion.minor) {
-        migrated += 1;
-      }
-    }
-    flutterUsage.sendEvent(kNullSafetyCategory, 'runtime-mode', label: nullSafetyMode.toString());
-    flutterUsage.sendEvent(kNullSafetyCategory, 'stats', parameters: CustomDimensions(
-      nullSafeMigratedLibraries: migrated,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/reporting/events.dart:294]: Can these fields be removed from the `CustomDimensions` class?

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/reporting/events.dart:294]: ooh, good catch

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 3 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/artifacts.dart:94]: Yeah, these are the unsound parts of the web sdk, which are no longer reachable.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/artifacts.dart:94]: This https://github.com/flutter/flutter/issues/122215 is tracking not building it

File: packages/flutter_tools/lib/src/reporting/events.dart (hunk 4 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/reporting/events.dart:294]: WDYT about this diff https://github.com/flutter/flutter/pull/120936/commits/3519e1f387ce6c7e46412f1fbd65c1cc00015223

File: packages/flutter_tools/lib/src/reporting/custom_dimensions.dart [hunk 5]
```diff
@@ -118,8 +116,8 @@ class CustomDimensions {
   final String? commandPackagesAndroidEmbeddingVersion;  // cd46
   final bool? nullSafety;  // cd47
   final bool? fastReassemble;  // cd48
-  final int? nullSafeMigratedLibraries;  // cd49
-  final int? nullSafeTotalLibraries;  // cd50
+  // final int? nullSafeMigratedLibraries;  // cd49
+  // final int? nullSafeTotalLibraries;  // cd50
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/reporting/custom_dimensions.dart:120]: Note the deprecation here like you do below.

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/lib/src/reporting/custom_dimensions.dart (hunk 5 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/reporting/custom_dimensions.dart]: done

File: packages/flutter/test_private/bin/test_private.dart [hunk 6]
```diff
@@ -225,7 +225,7 @@ class TestCase {
     for (final File test in tests) {
       final String testPath = path.join(path.dirname(test.path), 'lib', path.basenameWithoutExtension(test.path));
       final ProcessRunnerResult result = await runner.runProcess(
-        <String>[flutter, 'test', '--enable-experiment=non-nullable', '--null-assertions', testPath],
+        <String>[flutter, 'test', '--enable-experiment=non-nullable', testPath],
```
Comment (nshahan) [inline packages/flutter/test_private/bin/test_private.dart:228]: Is that non-nullable experiment flag used anymore? If this is the Dart SDK experiment flag it is no longer needed.
